pystac/catalog.py

```diff
@@ -739,9 +748,20 @@ def save(self, catalog_type: Optional[CatalogType] = None) -> None:
 
         for item_link in self.get_item_links():
             if item_link.is_resolved():
-                cast(pystac.Item, item_link.target).save_object(
-                    include_self_link=items_include_self_link
-                )
+                item = cast(pystac.Item, item_link.target)
+                item_dest_href = None
+                if dest_href is not None and root.catalog_type in [
+                    CatalogType.ABSOLUTE_PUBLISHED
+                ]:
+                    rel_href = make_relative_href(
+                        item.get_self_href(), self.get_self_href()
+                    )
```

Comment:
Since `get_self_href` can return `None`, we'll want to check for that case and raise an exception.

Reply:
I replaced calls to `get_self_href()` with `self_href`, which raises an exception in case of `None` value